### ISSUE TITLE
Performance regression in Might and Magic Heroes 7 returned since ff84448 ("Ubisoft-Connect" version). Works fine with 500db71.

### ISSUE THREAD

**Issue by Kerrung** [Frogging-Family/wine-tkg-git#426]
For this game I use Wine-tkg-git with Staging. All works fine with https://github.com/wine-staging/wine-staging/commit/fd5866f6f12e82afc74a21e6f39e65e117d250f2 (https://github.com/wine-mirror/wine/commit/1de583a4dac7d704b2d4291ada4a1885cd8cd1c9) and I have about 50 fps with ultra graphic settings. But I have about 3-15 fps since https://github.com/wine-staging/wine-staging/commit/9c8608eea6e0d15b84d4eec042dcd61d73c44fa3 or with newer Wine Staging commits. That is, I tried to do downgrade to https://github.com/wine-staging/wine-staging/commit/fd5866f6f12e82afc74a21e6f39e65e117d250f2, which solved this problem.
I haven't tried to test with vanilla Wine the latest git commit without Staging, so I'm not sure yet if it's vanilla's or Staging's regression.

OS: Arch Linux with GNOME;
GPU: Radeon RX 550
wine-tkg-git https://github.com/Frogging-Family/wine-tkg-git/commit/76a7fd23ba6366368002c840a51e984af824bad3
Might and Magic Heroes 7 use Unreal Engine 3 with Direct3D 9, so I use https://github.com/doitsujin/dxvk/commit/8a93bbd8faedf933e727581dc23f5a41eaeecad3

[last_build_config.log](https://github.com/Frogging-Family/wine-tkg-git/files/6696347/last_build_config.log)
[prepare.log](https://github.com/Frogging-Family/wine-tkg-git/files/6696348/prepare.log)
[mmh7-performance-regression.log](https://github.com/Frogging-Family/wine-tkg-git/files/6696726/mmh7-performance-regression.log)

**Comment by Kerrung** [comment]
I just retested. All works fine now with the latest Wine git since 112db36f4f84dc124cc7739c9523992343f91882 wine-tkg-git commit. So I close it as full fixed now. Thank you!

**Comment by Kerrung** [comment]
@Tk-Glitch  This issue returned since ff8444815c6182e45a85aca9e897a8ada1f7d6ae. So I reopen it. Works fine with 500db71e086698b4929e12824aa97218692d7402.

**Comment by Tk-Glitch** [comment]
> @Tk-Glitch This issue returned since [ff84448](https://github.com/Frogging-Family/wine-tkg-git/commit/ff8444815c6182e45a85aca9e897a8ada1f7d6ae). So I reopen it. Works fine with [500db71](https://github.com/Frogging-Family/wine-tkg-git/commit/500db71e086698b4929e12824aa97218692d7402).

It makes no sense. ff84448 is after 500db71 to begin with, so let's assume you have inverted them. But still, neither of those commits are touching the wine side, and ff84448 doesn't even affect trees newer than 6.9.

Possible culprits would be https://github.com/Frogging-Family/wine-tkg-git/commit/112db36f4f84dc124cc7739c9523992343f91882 (which is an older commit than what you listed), https://github.com/Frogging-Family/wine-tkg-git/commit/f1f0a2c2e5014177ac1aaa7fd873e87d063276c7 (which is much newer) or upstream changes.

**Comment by Kerrung** [comment]
> > @Tk-Glitch This issue returned since [ff84448](https://github.com/Frogging-Family/wine-tkg-git/commit/ff8444815c6182e45a85aca9e897a8ada1f7d6ae). So I reopen it. Works fine with [500db71](https://github.com/Frogging-Family/wine-tkg-git/commit/500db71e086698b4929e12824aa97218692d7402).
> 
> It makes no sense. [ff84448](https://github.com/Frogging-Family/wine-tkg-git/commit/ff8444815c6182e45a85aca9e897a8ada1f7d6ae) is after [500db71](https://github.com/Frogging-Family/wine-tkg-git/commit/500db71e086698b4929e12824aa97218692d7402) to begin with, so let's assume you have inverted them. But still, neither of those commits are touching the wine side, and [ff84448](https://github.com/Frogging-Family/wine-tkg-git/commit/ff8444815c6182e45a85aca9e897a8ada1f7d6ae) doesn't even affect trees newer than 6.9.
> 
> Possible culprits would be [112db36](https://github.com/Frogging-Family/wine-tkg-git/commit/112db36f4f84dc124cc7739c9523992343f91882) (which is an older commit than what you listed), [f1f0a2c](https://github.com/Frogging-Family/wine-tkg-git/commit/f1f0a2c2e5014177ac1aaa7fd873e87d063276c7) (which is much newer) or upstream changes.

In all cases, I only used the newest Wine Staging master, but tried to compile each TKG commit one at a time, starting with the newest one, and then restart my system each time and launch the game to measure FPS changes in it. It took me a lot of time, and I did so until I found that TKG commit, with which the FPS in this game is normal. And this commit turned out to be exactly 500db71e086698b4929e12824aa97218692d7402, despite the fact that for reasons unknown to me, when I tried ff8444815c6182e45a85aca9e897a8ada1f7d6ae, I saw significant FPS drawdowns. I don't know how it works, but I came to this conclusion through practical experiments. I can provide any evidence, logs, screenshots or whatever you ask me to. I can also double-check to further exclude the possibility of my conclusions being erroneous.

**Comment by Tk-Glitch** [comment]
Could you provide a wine log for both runs (fine and broken), please?

**Comment by Kerrung** [comment]
> Could you provide a wine log for both runs (fine and broken), please?

You were right about the fact that I specified the wrong commits. Performance regression starts since f1f0a2c2e5014177ac1aaa7fd873e87d063276c7, but works fine with 3263cb71daadabe5b2775787cc75797145f6a671. I figured it out after a series of additional tests.

[f1f0a2c2e5014177ac1aaa7fd873e87d063276c7 mmh7-performance-regression.log](https://github.com/Frogging-Family/wine-tkg-git/files/6743465/BAD.f1f0a2c2e5014177ac1aaa7fd873e87d063276c7.mmh7-performance-regression.log)

[3263cb71daadabe5b2775787cc75797145f6a671 mmh7-normal-performance-with-old-commit.log](https://github.com/Frogging-Family/wine-tkg-git/files/6743475/3263cb71daadabe5b2775787cc75797145f6a671.mmh7-normal-performance-with-old-commit.log)
